File: Ramsha.Ui/src/app/models/statistics/supplierStatistics.ts
import { SalesPerformance } from "./statistic";

export type ProductStockDistribution = {
    label: string;
    value: number;
};


export type SupplierStatistics = {
    totalFulfillments: number;
    totalSuppliedProducts: number;
    totalRevenue: number;
    topSellingProduct: string
    productStockDistribution: ProductStockDistribution[];
    salesPerformance: SalesPerformance[];
};

File: Ramsha.Application/Dtos/Statistics/SupplierStatistics.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Ramsha.Application.Dtos.Statistics;

public record SupplierStatistics(
int TotalFulfillments,
int TotalSuppliedProducts,
decimal TotalRevenue,
string TopSellingProduct,
List<ProductStockDistribution> ProductStockDistribution,
List<SalesPerformance> SalesPerformance
);


public record ProductStockDistribution(
    string Label,
    int Value
);
